### ISSUE TITLE
Conditional build feature

### ISSUE THREAD

**Issue by zajca** [Block8/PHPCI#1011]
Hi, I'm thinking about switch from jenkins to phpci, because it has too much features that I don't need. But one think I'm missing and it's conditional build or build parameters.

My use case is that I want CI to manage deploy of applications using deployer. So I created plugin to run deployer but I want to rollback if there is problem or pass different options for build. 

Is there way how to pass frontend specified options to plugin, in best case when I click on build.

**Comment by mavimo** [comment]
@zajca you are right, but you can add parameters (eg branch/commit hash) if you invoke the build using REST API, so I suppose you can easily create a bash script (or create a UI ) to specify parameter for the build (but it also depends on what parameter you need to specify).

**Comment by zajca** [comment]
In jenkins now I have "Choice parameter", it's just array of string and name of the variable. And in build step I have bash switch like this:

```
case "$Deployer" in
"deploy-prod1")
    dep deploy prod1;
    ;;
    
"rollback-prod1")
    dep rollback prod1;
    ;;
    
"deploy-test")
    dep deploy test;
    ;;

"rollback-test")
    dep rollback test;
    ;;

*)
    echo 'ERR';
    ;;
esac
```

And when I trigger build in UI, I can choose which option I want and it's passed to bash as variable.
